app/src/containers/b2b/AccountMain.tsx

```diff
@@ -186,17 +192,19 @@ export default class AccountMain extends React.Component<RouteComponentProps<Acc
           this.setState({
             accountName: accounts.name,
             mainAccountName: accounts.name,
-            externalId: accounts['external-id'],
+            externalId: accounts._accountmetadata ? accounts._accountmetadata[0]['external-id'] : null,
             registrationNumber: accounts['registration-id'],
             isLoading: false,
             legalName: accounts['legal-name'],
             associates: accounts._associateroleassignments[0]._element ? accounts._associateroleassignments[0]._element.map(element => ({ associate: element._associate[0], roles: element._roleinfo[0] })) : [],
             status: accounts._statusinfo[0]._status[0].status,
             editSubAccountUri: accounts.self.uri,
+            editMetadataUri: accounts._accountmetadata ? accounts._accountmetadata[0].self.uri : null,
             selfSignUpCode: accounts._selfsignupinfo ? accounts._selfsignupinfo[0]['self-signup-code'] : '',
             userEmail: profile._myprofile[0]._primaryemail[0].email,
             addAssociateUri: accounts._associateroleassignments[0]._associateform[0]._addassociateaction[0].self.uri,
             addSubAccountUri: accounts._subaccounts[0]._accountform[0].self.uri,
+            addSubAccountSellerAdmin: typeof accounts._subaccounts[0]._accountform[0]['external-id'] !== 'undefined',
```

Comment:
Did you intend to get the type of this object here, or did you want the actual object?

Reply:
Did you test if no subaccounts are returned? Does this fail?

Reply:
this is just checking if on the `accountform` there is external-id field. sellers will see this on their form, buyers will not.
this will work without subaccounts, because it's not dealing with its elements, the form will always be present on `subaccounts` link in order to create new ones 😄

Reply:
Discussed in person. The intention makes sense.